### PR TITLE
add latex linter and linter action

### PR DESCRIPTION
**Pull request by kyle4908**
Some lines are well over 80 characters, it's not a hard limit, but we don't wanna go to far over, so I added this formatter which checks wraps (also check other things, but doesn't matter that much). You'll need cargo and then to follow the steps in the action to run it locally.

### CODE REVIEW

File: .github/workflows/latex-linter.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  pull_request:
+    types: [opened, reopened, synchronize]
+    paths: ["docs/**.tex"]
+jobs:
+  run-linter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - run: cargo install tex-fmt
```
Comment (kyle4908) [inline .github/workflows/latex-linter.yaml:12]: kind of bad practice at some point I'll initialize this repo as a rust project with a dummy, lib.rs or main.rs, and make a lockfile to install the dependencies from, but I'm being a bit lazy rn

File: .github/workflows/latex-linter.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    types: [opened, reopened, synchronize]
+    paths: ["docs/**/**.tex"]
+jobs:
+  run-linter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - run: cargo install tex-fmt
+      - run: tex-fmt --check ./docs/**/**.tex
```
Comment (kyle4908) [inline .github/workflows/latex-linter.yaml:13]: aight I'm prolly dumb, but when I did ./docs/**.tex it didn't check subdirectories, so maybe I'm a noob, but Idk I think I'm just gonna stick with this and we can fix it later

File: docs/SRS-Volere/SRS.tex
Comment (kyle4908) [inline docs/SRS-Volere/SRS.tex:1]: just ignore all changes except for this file, you're not reviewing all that

Comment (kyle4908) [reply docs/SRS-Volere/SRS.tex:1]: this is also the only file we've edited so far, so the rest aren't gonna get checked

File: .github/workflows/latex-linter.yaml (hunk 1 above)
Comment (kyle4908) [reply .github/workflows/latex-linter.yaml:12]: also this stuff is cached by the rust toolchain action, so don't worry about reinstall every time

**Review by MitchHynes — APPROVED** [review]

**Pull request merged**